lnwire/lnwire.go

```diff
@@ -854,10 +859,10 @@ func ReadElement(r io.Reader, element interface{}) error {
 
 // ReadElements deserializes a variable number of elements into the passed
 // io.Reader, with each element being deserialized according to the ReadElement
-// function.
-func ReadElements(r io.Reader, elements ...interface{}) error {
+// function. The passed protocol version may affect how the items are encoded.
+func ReadElements(r io.Reader, pver uint32, elements ...interface{}) error {
```

Comment:
If I understand it correctly, the propagation of `pver` and creation of a new version is only required for the migrations and a new node will never use the legacy version. I don't understand why we don't continue the same reasoning as before and copy over `lnwire`. In the past it was said that the copy is cheap and safe and not an issue. With the currently proposed change, we will always need to be careful not to break a migration still.

Reply:
You're suggesting copying over the entirety of `lnwire` for this minor change to conditionally continue to read bytes at the end of a message (the way it should've been all along really)? I don't think the bloat is worth it, the migration tests should also be such that if this section does break, then a migration test breaks as it runs into an EOF error.